api/admin.go

```diff
@@ -177,10 +177,6 @@ func (a *API) adminUserCreate(w http.ResponseWriter, r *http.Request) error {
 	ctx := r.Context()
 	config := a.getConfig(ctx)
 
-	if config.DisableSignup {
-		return forbiddenError("Signups not allowed for this instance")
-	}
-
```

Comment:
Just to double-check - is any of this functionality/route used by the "users"? eg, we don't need to do `if (!admin && config.DisableSignup)` instead.

I assume not, if all the tests are passing, but want to double-check

Reply:
yea this logic can only be reached with the service_role key

Reply:
yup, we protect all `/admin` routes with a [middleware](https://github.com/supabase/gotrue/blob/aead739cfff919500cdee343496279743448bca8/api/api.go#L151) that checks for the service role key